File: backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/ManagementData.java
package de.bund.digitalservice.ris.caselaw.domain;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Future;
import jakarta.validation.constraints.PastOrPresent;
import java.time.Instant;
import java.time.LocalDateTime;
import java.util.List;
import lombok.Builder;

@Builder(toBuilder = true)
public record ManagementData(
    @PastOrPresent LocalDateTime lastHandoverDateTime,
    @Future LocalDateTime scheduledPublicationDateTime,
    @Email String scheduledByEmail,
    List<DuplicateRelation> duplicateRelations,
    List<String> borderNumbers,
    Instant lastUpdatedAtDateTime,
    String lastUpdatedByName,
    String lastUpdatedByDocOffice,
    Instant createdAtDateTime,
    String createdByName,
    String createdByDocOffice,
    Instant firstPublishedAtDateTime,
    Instant lastPublishedAtDateTime) {}

File: frontend/src/domain/managementData.ts
import { PublicationState } from "@/domain/publicationStatus"

export enum DuplicateRelationStatus {
  PENDING = "PENDING",
  IGNORED = "IGNORED",
}

export type DuplicateRelation = {
  documentNumber: string
  status: DuplicateRelationStatus
  isJdvDuplicateCheckActive: boolean
  courtLabel?: string
  decisionDate?: string
  fileNumber?: string
  documentType?: string
  publicationStatus?: PublicationState
}

export type ManagementData = {
  scheduledPublicationDateTime?: string
  scheduledByEmail?: string
  duplicateRelations: DuplicateRelation[]
  borderNumbers: string[]
  lastUpdatedAtDateTime?: string
  lastUpdatedByName?: string
  lastUpdatedByDocOffice?: string
  createdAtDateTime?: string
  createdByName?: string
  createdByDocOffice?: string
  firstPublishedAtDateTime?: string
}
